cli/deployment/deployment.go

```diff
@@ -52,30 +60,43 @@ func (d Deployment) FullName() string {
 	return d.config.FullName()
 }
 
-func replaceOutRefs(config Config, outputs map[string]map[string]string) []byte {
-	data, err := config.YAML()
-	if err != nil {
-		log.Fatalf("error while parsing yaml for config: %s, error: %v", config.FullName(), err)
+func (d *Deployment) Execute(action string, preview bool) (output string, error error) {
```

Comment:
Please add a doc comment for this exported method as described at https://golang.org/doc/effective_go.html#commentary